zerver/data_import/slack.py

```diff
@@ -594,19 +593,27 @@ def channel_message_to_zerver_message(realm_id: int, users: List[ZerverFieldsT],
                                                 zerver_realmemoji)
 
         # Process different subtypes of slack messages
-        if 'subtype' in message.keys():
-            subtype = message['subtype']
-            # Subtypes which have only the action in the message should
-            # be rendered with '/me' in the content initially
-            # For example "sh_room_created" has the message 'started a call'
-            # which should be displayed as '/me started a call'
-            if subtype in ["bot_add", "sh_room_created", "me_message"]:
-                content = ('/me %s' % (content))
 
-            # For attachments with slack download link
-            elif subtype == "file_share" and 'files.slack.com' in message['file']['url_private']:
+        # Subtypes which have only the action in the message should
+        # be rendered with '/me' in the content initially
+        # For example "sh_room_created" has the message 'started a call'
+        # which should be displayed as '/me started a call'
+        if subtype in ["bot_add", "sh_room_created", "me_message"]:
+            content = ('/me %s' % (content))
+
+        files = message.get('files', False)
+        if subtype == 'file_share' or files:
+            # In the messages, uploads can either have the subtype as 'file_share' or
+            # have the upload information in 'files' keyword
+            if subtype == 'file_share':
                 fileinfo = message['file']
+                url = message['file']['url_private']
+            elif files:
+                fileinfo = files
+                url = files['url_private']
```

Comment:
`files` is a list of objects, in this case.  So I think this needs the following patch:

```
diff --git a/zerver/data_import/slack.py b/zerver/data_import/slack.py
index bd7a7d3..9cfde68 100755
--- a/zerver/data_import/slack.py
+++ b/zerver/data_import/slack.py
@@ -601,17 +601,14 @@ def channel_message_to_zerver_message(realm_id: int, users: List[ZerverFieldsT],
         if subtype in ["bot_add", "sh_room_created", "me_message"]:
             content = ('/me %s' % (content))
 
-        files = message.get('files', False)
-        if subtype == 'file_share' or files:
-            # In the messages, uploads can either have the subtype as 'file_share' or
+        files = message.get('files', [])
+        if subtype == 'file_share':
+            # In Slack messages, uploads can either have the subtype as 'file_share' or
             # have the upload information in 'files' keyword
-            if subtype == 'file_share':
-                fileinfo = message['file']
-                url = message['file']['url_private']
-            elif files:
-                fileinfo = files
-                url = files['url_private']
+            files = [message['file']]
 
+        for fileinfo in files:
+            url = fileinfo['url_private']
             # For attachments with slack download link
             if 'files.slack.com' in url:
                 has_attachment = has_link = True
```